### PR TITLE
address deprecated calls in otelgrpc instrumentation

### PR DESCRIPTION
**Pull request by codeboten**

### CODE REVIEW

File: config/configgrpc/configgrpc.go [hunk 1]
```diff
@@ -259,8 +259,7 @@ func (gcs *GRPCClientSettings) toDialOptions(host component.Host, settings compo
 	}
 
 	// Enable OpenTelemetry observability plugin.
-	opts = append(opts, grpc.WithUnaryInterceptor(otelgrpc.UnaryClientInterceptor(otelOpts...)))
-	opts = append(opts, grpc.WithStreamInterceptor(otelgrpc.StreamClientInterceptor(otelOpts...)))
+	opts = append(opts, grpc.WithStatsHandler(otelgrpc.NewClientHandler(otelOpts...)))
```
Comment (bogdandrutu) [inline config/configgrpc/configgrpc.go:262]: Finally they did it.

**Review by bogdandrutu — APPROVED** [review]

File: config/configgrpc/configgrpc.go (hunk 1 above)
Comment (codeboten) [reply config/configgrpc/configgrpc.go:262]: Yeah although based on the results, something doesn't look right: https://github.com/open-telemetry/opentelemetry-collector/actions/runs/6853173539/job/18633503149?pr=8855

Comment (codeboten) [reply config/configgrpc/configgrpc.go:262]: @bogdandrutu ah nevermind, i see the test is just checking the len of the opts 🤣 will submit a PR to fix it

Comment (codeboten) [reply config/configgrpc/configgrpc.go:262]: See https://github.com/open-telemetry/opentelemetry-collector/pull/8864

**Pull request merged**